### ISSUE TITLE
Some fields should not be translated by the JS

### ISSUE THREAD

**Issue by herrvigg** [qtranslate/qtranslate-xt#61]
<a href="https://github.com/soft79"><img src="https://avatars2.githubusercontent.com/u/7942726?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [soft79](https://github.com/soft79)**
_Sunday Mar 01, 2015 at 19:09 GMT_
_Originally opened as https://github.com/qTranslate-Team/qtranslate-x/issues/61_

----

Creating a filter to exclude certain fields from the JS translation buttons would be very useful. 

Please see https://github.com/josk79/qtranslate-x/pull/1

This adds an option to the admin page that allows for CSS-selectors for DOM-elements that should not be translated. 

This branch also applies a filter before passing the config to JavaScript. This way plugins can add certain filters. For example: "WooCommerce qTranslate X" should add `body.post-type-shop_coupon #title` because coupon titles shouldn't be translated as it breaks functionality.

**Comment by herrvigg** [comment]
<a href="https://github.com/soft79"><img src="https://avatars2.githubusercontent.com/u/7942726?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [soft79](https://github.com/soft79)**
_Sunday Mar 01, 2015 at 19:30 GMT_

----

Example of the filter hook:

```
//Don't translate WooCommerce coupon title
add_filter('qtranxf_get_js_config', function($config) { 
    $config['excluded_field_selectors'][] = "body.post-type-shop_coupon #title";
    return $config;
} );
```

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Sunday Mar 01, 2015 at 23:06 GMT_

----

Hi @josk79, Why do we need to exclude something if we can not to include it to begin with? Which page/field this exclusion applies to?

**Comment by herrvigg** [comment]
<a href="https://github.com/soft79"><img src="https://avatars2.githubusercontent.com/u/7942726?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [soft79](https://github.com/soft79)**
_Sunday Mar 01, 2015 at 23:34 GMT_

----

In this case the title field of a post with type shop_coupon .

If a user enters a coupon code in WooCommerce, the coupon is looked up by a `SELECT * FROM $wp->db->posts WHERE title = %s` query

There is a filter to alter this query, but the title is not passed as an argument, so it would be hard to fix it this way, that's way I came up with the exclude filter. might be useful for other custom post types as well..

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Monday Mar 02, 2015 at 02:45 GMT_

----

If there was no a problem, would you want to make coupon translatable?

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Monday Mar 02, 2015 at 04:59 GMT_

----

BTW, @josk79, if you set the same value for all languages, then language tags should be removed. This trick should enable you to keep going until we figure out the best way to solve this problem.

**Comment by herrvigg** [comment]
<a href="https://github.com/soft79"><img src="https://avatars2.githubusercontent.com/u/7942726?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [soft79](https://github.com/soft79)**
_Monday Mar 02, 2015 at 08:34 GMT_

----

For me a coupon title doesn't need to be translatable, only it's excerpt. I'll try the workaround tonight.

**Comment by herrvigg** [comment]
<a href="https://github.com/soft79"><img src="https://avatars2.githubusercontent.com/u/7942726?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [soft79](https://github.com/soft79)**
_Monday Mar 02, 2015 at 19:26 GMT_

----

Workaround seems to work. Sometimes the solutions can be so simple... :)

**Comment by herrvigg** [comment]
<a href="https://github.com/mweimerskirch"><img src="https://avatars3.githubusercontent.com/u/362092?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mweimerskirch](https://github.com/mweimerskirch)**
_Wednesday Mar 04, 2015 at 21:17 GMT_

----

I like the idea of being able to add exceptions. We use lots of custom post types for all sorts of data. Some of them do not require the title field to be translatable. Being able to add exceptions via hooks would be really helpful in these cases.

**Comment by herrvigg** [comment]
<a href="https://github.com/realtebo"><img src="https://avatars2.githubusercontent.com/u/1268699?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [realtebo](https://github.com/realtebo)**
_Friday Dec 16, 2016 at 15:44 GMT_

----

I've the same need, in our specific situation we need to avoid to make translatable the post title for a custom post type. 

I need a workaround

**Comment by herrvigg** [comment]
<a href="https://github.com/lmartins"><img src="https://avatars2.githubusercontent.com/u/151981?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [lmartins](https://github.com/lmartins)**
_Monday May 01, 2017 at 10:35 GMT_

----

Also looking for a solution to this, any progress on this since last year?

**Comment by herrvigg** [comment]
<a href="https://github.com/jbvidex"><img src="https://avatars0.githubusercontent.com/u/9750924?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jbvidex](https://github.com/jbvidex)**
_Friday Jan 12, 2018 at 22:00 GMT_

----

+1
I have a custom post type for which I do not want the title field (part numbers, in our case) to be translated. The language tags are messing up my Next/Previous post navigation links. :(

**Comment by herrvigg** [comment]
Fixed by #883.
